### PR TITLE
fix: Make generate-legacy-interop-components script take formatting into account

### PR DESCRIPTION
**Pull request by okwasniewski**
## Summary:

When Codegen generates interop components it reformats `RCTLegacyInteropComponents.mm` to have each component in a new line. This creates this unnecessary diff every time I run `pod install`. This PR aims to commit a formatted version of the codegen-generated file to avoid creating this diff every time. Here is a reference of Codegen function generating this: https://github.com/facebook/react-native/blob/44d6e4310cc9ad0d711d05e8dd5ee5220738e5b5/packages/react-native/scripts/codegen/generate-legacy-interop-components.js#L53

CleanShot 2023-12-12 at 14 09 34@2x

## Changelog:

[INTERNAL] [CHANGED] - Reformat `RCTLegacyInteropComponents` to follow Codegen generated formatting

## Test Plan:

CI Green

### CODE REVIEW

**Review by cortinico — APPROVED** [review]

**Comment by cortinico** [comment]
@okwasniewski so if I import this, the internal formatter will attempt to reformat it back to the original code

**Comment by okwasniewski** [comment]
@cortinico Thanks for letting me know. I've modified the Codegen script to take into account Meta internal linter (I wasn't sure if the split should happen at 3 or 4 lines) but now the linter should be happy and it shouldn't generate unnecessary diff every time

File: packages/react-native/Libraries/AppDelegate/RCTLegacyInteropComponents.mm [hunk 1]
```diff
@@ -11,7 +11,7 @@ @implementation RCTLegacyInteropComponents
 
 + (NSArray<NSString *> *)legacyInteropComponents
 {
-  return @[ @"RNTMyLegacyNativeView", @"RNTMyNativeView" ];
+  return @[ @"RNTMyLegacyNativeView", @"RNTMyNativeView"];
```
Comment (cortinico) [inline packages/react-native/Libraries/AppDelegate/RCTLegacyInteropComponents.mm:14]: can we undo this change also?

**Review by cortinico — CHANGES_REQUESTED** [review]

File: packages/react-native/Libraries/AppDelegate/RCTLegacyInteropComponents.mm (hunk 1 above)
Comment (okwasniewski) [reply packages/react-native/Libraries/AppDelegate/RCTLegacyInteropComponents.mm]: Removed

File: packages/react-native/scripts/codegen/generate-legacy-interop-components.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ const PROJECT_FIELD = 'project';
 const IOS_FIELD = 'ios';
 const LEGACY_COMPONENTS_FIELD = 'unstable_reactLegacyComponentNames';
 const OUTPUT_FILE_NAME = 'RCTLegacyInteropComponents.mm';
+const COMPONENTS_IN_ONE_LINE = 3;
```
Comment (cipolleschi) [inline packages/react-native/scripts/codegen/generate-legacy-interop-components.js:21]: more than the number of components, it is a matter of number of chars.
I think that the linter will go to a new line if we overflow the 80 or 120 char mark.

Could we refactor the PR with a char-based logic? Because if I have 4 short components:

```
@"RCTA", @"RCTB", @"RCTC", @"RCTD",   
``` 
the internal linter will not put them on multiple lines. :P

**Review by cipolleschi — CHANGES_REQUESTED** [review]

File: packages/react-native/scripts/codegen/generate-legacy-interop-components.js (hunk 2 above)
Comment (cortinico) [reply packages/react-native/scripts/codegen/generate-legacy-interop-components.js:21]: great point 👍

Comment (okwasniewski) [reply packages/react-native/scripts/codegen/generate-legacy-interop-components.js]: Indeed great point! I've adjusted the implementation 🙌
